src/rules/flake8_executable/rules/shebang_whitespace.rs

```diff
@@ -0,0 +1,51 @@
+use ruff_macros::derive_message_formats;
+use rustpython_ast::Location;
+
+use crate::ast::types::Range;
+use crate::define_violation;
+use crate::fix::Fix;
+use crate::registry::Diagnostic;
+use crate::rules::flake8_executable::helpers::{extract_shebang, ShebangDirective};
+use crate::violation::AlwaysAutofixableViolation;
+
+define_violation!(
+    pub struct ShebangWhitespace;
+);
+impl AlwaysAutofixableViolation for ShebangWhitespace {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("There is whitespace before the shebang.")
```

Comment:
Sorry for the nit, but we tend to exclude punctuation unless it's a multi-sentence message. Stylistically, this might fit in a little more: `format!("Avoid whitespace before shebang")`